linera-base/build.rs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::io::BufRead as _;
+
+fn main() {
+    let mut versions = std::process::Command::new("bash")
+        .arg("versions.sh")
+        .stdout(std::process::Stdio::piped())
+        .spawn()
+        .expect("failed to launch child");
```

Comment:
`jq` is already a requirement (for testing, if not building), so I didn't have an issue with including it.  `sha256sum` is part of coreutils so I expected it to be available everywhere.  [The internet](https://apple.stackexchange.com/a/230920) says we can use `shasum -a 256` instead; can you check this works & has the ‘same’ output (hash as lowercase hexadecimal in the first column)?

We could save the SHA at the same time, but it seems more fragile (possibilities for the hash to get out of sync with the code, et cetera).  The performance here is sufficient that I'm not really concerned about precalculating it and I think it's not worth the cost in terms of moving parts.  It also adds yet another thing that we'll forget to update when making PRs, which is a hit to DX.  The ‘fire and forget’ flow with CI is already suffering a fair bit from these.  I also don't love having build scripts (since they're Cargo-specific, and running arbitrary code during a build seems messy) but I do prefer to keep all this logic in one place.

There is `CARGO_PKG_VERSION` but I thought it would be nice to have all the version info generated together :)

I considered porting this script to Rust (putting it directly in the build script) but I'm not sure if it's worth it.  It would be significantly longer and less readable, even with `vergen`.   Maybe if we're not happy to depend on `jq` and `sha256sum` it might be worthwhile.

Reply:
Apropos of offline conversation: I've removed usage of `jq` (replaced by using `CARGO_PKG_VERSION`) and added a fallback to use `shasum -a 256` if `sha256sum` isn't available.